augur/application/cli/backend.py

```diff
@@ -162,6 +140,51 @@ def start(disable_collection, development, port):
             except RedisConnectionError:
                 pass
 
+def start_celery_worker_processes():
+
+    #Calculate process scaling based on how much memory is available on the system in bytes.
+    #Each celery process takes ~500MB or 500 * 1024^2 bytes
+
+    #Cap memory usage to 30% of total virtual memory
+    available_memory_in_bytes = psutil.virtual_memory().total * .3
+    available_memory_in_megabytes = available_memory_in_bytes / (1024 ** 2)
+    max_process_estimate = available_memory_in_megabytes // 500
+
+    #Get a subset of the maximum procesess available using a ratio, not exceeding a maximum value
+    def determine_worker_processes(ratio,maximum):
+        return min(round(max_process_estimate * ratio),maximum)
+
```

Comment:
We need to either add a min of 1 to this, or return an error if that the system does not have enough memory to run Augur because a system with 8 GB of RAM with Augur configured to use 30 percent of virtual memory will currently run 2 scheduling processes, 1 core process, 0 secondary processes, and 0 facade processes. Note: Around 8.2 GB with a 30 percent usage is what is needed to for 2 scheduling processes and 1 of all the others. At 20 percent usage 12.3 GB is needed

Reply:
I think we should bump it up to at least 40% and then add a minimum process of 1.

Reply:
We already have a hard limit on the maximum level of processes each worker can have for larger instances. I think for now a default cap of 20-30% would be fine. I think it is a good idea to put the percentage in the config though.

Reply:
Yeah I didn't notice that before, this sounds good to me